google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -213,13 +214,18 @@ public B setHost(String host) {
     }
 
     /**
-     * Sets the service authentication credentials.
+     * Sets the service authentication credentials. If this method or {@link #setNoCredentials() are
+     * not used on the builder, {@link GoogleCredentials#getApplicationDefault()} will be used to
+     * attempt getting credentials from the environment.
      *
+     * @param credentials authentication credentials, should not be {@code null}
      * @return the builder
+     * @throws NullPointerException if {@code credentials} is {@code null}. To disable
+     *     authentication use {@link Builder#setNoCredentials()}
```

Comment:
Given that we now throw on `null`:
```java
options.toBuilder().setCredentials(GoogleCredentials.getApplicationDefault()).build();
```
If we revert the NPE on `null` you could also do the following (where `build()` does the discovery for you):
```java
options.toBuilder().setCredentials(null).build();
```
I have no personal preference, if you do I am happy to go with it.